### ISSUE TITLE
qwen3-embedding-0.6b部署后无法使用

### ISSUE THREAD

**Issue by wuyabing** [gpustack/gpustack#2171]
运行的qwen3-embedding-0.6b ，在试验场使用时候，报如下错误：The model does not support Embeddings API

**Comment by pengjiang80** [comment]
Please provide the complete log of the model instance.

**Comment by wuyabing** [comment]
完整日志如下：
2025-06-06T09:50:15+00:00 - gpustack.worker.downloaders - INFO - Retriving file lock
Downloading Model to directory: /var/lib/gpustack/cache/model_scope/Qwen/Qwen3-Embedding-0.6B
Downloading Model to directory: /root/.cache/modelscope/hub/Qwen/Qwen3-Embedding-0.6B
2025-06-06T09:50:18+00:00 - gpustack.worker.backends.vllm - INFO - Starting vllm server
INFO 06-06 09:50:20 __init__.py:190] Automatically detected platform cuda.
INFO 06-06 09:50:21 api_server.py:840] vLLM API server version 0.7.2
INFO 06-06 09:50:21 api_server.py:841] args: Namespace(subparser='serve', model_tag='/var/lib/gpustack/cache/model_scope/Qwen/Qwen3-Embedding-0___6B', config='', host='0.0.0.0', port=40682, uvicorn_log_level='info', allow_credentials=False, allowed_origins=['*'], allowed_methods=['*'], allowed_headers=['*'], api_key=None, lora_modules=None, prompt_adapters=None, chat_template=None, chat_template_content_format='auto', response_role='assistant', ssl_keyfile=None, ssl_certfile=None, ssl_ca_certs=None, ssl_cert_reqs=0, root_path=None, middleware=[], return_tokens_as_token_ids=False, disable_frontend_multiprocessing=False, enable_request_id_headers=False, enable_auto_tool_choice=False, enable_reasoning=False, reasoning_parser=None, tool_call_parser=None, tool_parser_plugin='', model='/var/lib/gpustack/cache/model_scope/Qwen/Qwen3-Embedding-0___6B', task='auto', tokenizer=None, skip_tokenizer_init=False, revision=None, code_revision=None, tokenizer_revision=None, tokenizer_mode='auto', trust_remote_code=False, allowed_local_media_path=None, download_dir=None, load_format='auto', config_format=<ConfigFormat.AUTO: 'auto'>, dtype='auto', kv_cache_dtype='auto', max_model_len=8192, guided_decoding_backend='xgrammar', logits_processor_pattern=None, model_impl='auto', distributed_executor_backend=None, pipeline_parallel_size=1, tensor_parallel_size=1, max_parallel_loading_workers=None, ray_workers_use_nsight=False, block_size=None, enable_prefix_caching=None, disable_sliding_window=False, use_v2_block_manager=True, num_lookahead_slots=0, seed=0, swap_space=4, cpu_offload_gb=0, gpu_memory_utilization=0.2, num_gpu_blocks_override=None, max_num_batched_tokens=None, max_num_seqs=None, max_logprobs=20, disable_log_stats=False, quantization=None, rope_scaling=None, rope_theta=None, hf_overrides=None, enforce_eager=False, max_seq_len_to_capture=8192, disable_custom_all_reduce=False, tokenizer_pool_size=0, tokenizer_pool_type='ray', tokenizer_pool_extra_config=None, limit_mm_per_prompt=None, mm_processor_kwargs=None, disable_mm_preprocessor_cache=False, enable_lora=False, enable_lora_bias=False, max_loras=1, max_lora_rank=16, lora_extra_vocab_size=256, lora_dtype='auto', long_lora_scaling_factors=None, max_cpu_loras=None, fully_sharded_loras=False, enable_prompt_adapter=False, max_prompt_adapters=1, max_prompt_adapter_token=0, device='auto', num_scheduler_steps=1, multi_step_stream_outputs=True, scheduler_delay_factor=0.0, enable_chunked_prefill=None, speculative_model=None, speculative_model_quantization=None, num_speculative_tokens=None, speculative_disable_mqa_scorer=False, speculative_draft_tensor_parallel_size=None, speculative_max_model_len=None, speculative_disable_by_batch_size=None, ngram_prompt_lookup_max=None, ngram_prompt_lookup_min=None, spec_decoding_acceptance_method='rejection_sampler', typical_acceptance_sampler_posterior_threshold=None, typical_acceptance_sampler_posterior_alpha=None, disable_logprobs_during_spec_decoding=None, model_loader_extra_config=None, ignore_patterns=[], preemption_mode=None, served_model_name=['qwen3-embedding-0.6b'], qlora_adapter_name_or_path=None, otlp_traces_endpoint=None, collect_detailed_traces=None, disable_async_output_proc=False, scheduling_policy='fcfs', override_neuron_config=None, override_pooler_config=None, compilation_config=None, kv_transfer_config=None, worker_cls='auto', generation_config=None, override_generation_config=None, enable_sleep_mode=False, calculate_kv_scales=False, disable_log_requests=False, max_log_len=None, disable_fastapi_docs=False, enable_prompt_tokens_details=False, dispatch_function=<function serve at 0x7efef0a41e10>)
INFO 06-06 09:50:21 api_server.py:206] Started engine process with PID 38184
INFO 06-06 09:50:24 __init__.py:190] Automatically detected platform cuda.
INFO 06-06 09:50:26 config.py:542] This model supports multiple tasks: {'reward', 'embed', 'classify', 'generate', 'score'}. Defaulting to 'generate'.
INFO 06-06 09:50:29 config.py:542] This model supports multiple tasks: {'embed', 'reward', 'classify', 'generate', 'score'}. Defaulting to 'generate'.
INFO 06-06 09:50:29 llm_engine.py:234] Initializing a V0 LLM engine (v0.7.2) with config: model='/var/lib/gpustack/cache/model_scope/Qwen/Qwen3-Embedding-0___6B', speculative_config=None, tokenizer='/var/lib/gpustack/cache/model_scope/Qwen/Qwen3-Embedding-0___6B', skip_tokenizer_init=False, tokenizer_mode=auto, revision=None, override_neuron_config=None, tokenizer_revision=None, trust_remote_code=False, dtype=torch.bfloat16, max_seq_len=8192, download_dir=None, load_format=auto, tensor_parallel_size=1, pipeline_parallel_size=1, disable_custom_all_reduce=False, quantization=None, enforce_eager=False, kv_cache_dtype=auto,  device_config=cuda, decoding_config=DecodingConfig(guided_decoding_backend='xgrammar'), observability_config=ObservabilityConfig(otlp_traces_endpoint=None, collect_model_forward_time=False, collect_model_execute_time=False), seed=0, served_model_name=qwen3-embedding-0.6b, num_scheduler_steps=1, multi_step_stream_outputs=True, enable_prefix_caching=False, chunked_prefill_enabled=False, use_async_output_proc=True, disable_mm_preprocessor_cache=False, mm_processor_kwargs=None, pooler_config=None, compilation_config={"splitting_ops":[],"compile_sizes":[],"cudagraph_capture_sizes":[256,248,240,232,224,216,208,200,192,184,176,168,160,152,144,136,128,120,112,104,96,88,80,72,64,56,48,40,32,24,16,8,4,2,1],"max_capture_size":256}, use_cached_outputs=True, 
INFO 06-06 09:50:30 cuda.py:230] Using Flash Attention backend.
INFO 06-06 09:50:50 model_runner.py:1110] Starting to load model /var/lib/gpustack/cache/model_scope/Qwen/Qwen3-Embedding-0___6B...
WARNING 06-06 09:50:50 utils.py:69] Qwen3ForCausalLM has no vLLM implementation, falling back to Transformers implementation. Some features may not be supported and performance may not be optimal.
INFO 06-06 09:50:50 transformers.py:132] Using Transformers backend.
Loading safetensors checkpoint shards:   0% Completed | 0/1 [00:00<?, ?it/s]
Loading safetensors checkpoint shards: 100% Completed | 1/1 [00:00<00:00,  3.35it/s]
Loading safetensors checkpoint shards: 100% Completed | 1/1 [00:00<00:00,  3.35it/s]
INFO 06-06 09:50:51 model_runner.py:1115] Loading model weights took 1.1098 GB
INFO 06-06 09:50:52 worker.py:267] Memory profiling takes 0.53 seconds
INFO 06-06 09:50:52 worker.py:267] the current vLLM instance can use total_gpu_memory (44.31GiB) x gpu_memory_utilization (0.20) = 8.86GiB
INFO 06-06 09:50:52 worker.py:267] model weights take 1.11GiB; non_torch_memory takes 0.06GiB; PyTorch activation peak memory takes 1.40GiB; the rest of the memory reserved for KV Cache is 6.29GiB.
INFO 06-06 09:50:52 executor_base.py:110] # CUDA blocks: 3682, # CPU blocks: 2340
INFO 06-06 09:50:52 executor_base.py:115] Maximum concurrency for 8192 tokens per request: 7.19x
INFO 06-06 09:50:53 model_runner.py:1434] Capturing cudagraphs for decoding. This may lead to unexpected consequences if the model is not static. To run the model in eager mode, set 'enforce_eager=True' or use '--enforce-eager' in the CLI. If out-of-memory error occurs during cudagraph capture, consider decreasing `gpu_memory_utilization` or switching to eager mode. You can also reduce the `max_num_seqs` as needed to decrease memory usage.
Capturing CUDA graph shapes: 100%|██████████| 35/35 [00:16<00:00,  2.15it/s]
INFO 06-06 09:51:09 model_runner.py:1562] Graph capturing finished in 16 secs, took 0.59 GiB
INFO 06-06 09:51:09 llm_engine.py:431] init engine (profile, create kv cache, warmup model) took 18.44 seconds
INFO 06-06 09:51:10 api_server.py:756] Using supplied chat template:
INFO 06-06 09:51:10 api_server.py:756] None
INFO 06-06 09:51:10 launcher.py:21] Available routes are:
INFO 06-06 09:51:10 launcher.py:29] Route: /openapi.json, Methods: HEAD, GET
INFO 06-06 09:51:10 launcher.py:29] Route: /docs, Methods: HEAD, GET
INFO 06-06 09:51:10 launcher.py:29] Route: /docs/oauth2-redirect, Methods: HEAD, GET
INFO 06-06 09:51:10 launcher.py:29] Route: /redoc, Methods: HEAD, GET
INFO 06-06 09:51:10 launcher.py:29] Route: /health, Methods: GET
INFO 06-06 09:51:10 launcher.py:29] Route: /ping, Methods: POST, GET
INFO 06-06 09:51:10 launcher.py:29] Route: /tokenize, Methods: POST
INFO 06-06 09:51:10 launcher.py:29] Route: /detokenize, Methods: POST
INFO 06-06 09:51:10 launcher.py:29] Route: /v1/models, Methods: GET
INFO 06-06 09:51:10 launcher.py:29] Route: /version, Methods: GET
INFO 06-06 09:51:10 launcher.py:29] Route: /v1/chat/completions, Methods: POST
INFO 06-06 09:51:10 launcher.py:29] Route: /v1/completions, Methods: POST
INFO 06-06 09:51:10 launcher.py:29] Route: /v1/embeddings, Methods: POST
INFO 06-06 09:51:10 launcher.py:29] Route: /pooling, Methods: POST
INFO 06-06 09:51:10 launcher.py:29] Route: /score, Methods: POST
INFO 06-06 09:51:10 launcher.py:29] Route: /v1/score, Methods: POST
INFO 06-06 09:51:10 launcher.py:29] Route: /rerank, Methods: POST
INFO 06-06 09:51:10 launcher.py:29] Route: /v1/rerank, Methods: POST
INFO 06-06 09:51:10 launcher.py:29] Route: /v2/rerank, Methods: POST
INFO 06-06 09:51:10 launcher.py:29] Route: /invocations, Methods: POST
INFO:     Started server process [38163]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://0.0.0.0:40682 (Press CTRL+C to quit)
INFO:     127.0.0.1:34836 - "GET /v1/models HTTP/1.1" 200 OK
INFO:     192.168.72.215:55708 - "GET /v1/models HTTP/1.1" 200 OK
INFO:     192.168.72.215:56000 - "POST /v1/embeddings HTTP/1.1" 200 OK
INFO:     192.168.72.215:56040 - "POST /v1/embeddings HTTP/1.1" 200 OK
INFO:     192.168.72.215:56090 - "POST /v1/embeddings HTTP/1.1" 200 OK
INFO:     192.168.72.215:56128 - "POST /v1/embeddings HTTP/1.1" 200 OK
INFO:     192.168.72.215:56378 - "POST /v1/chat/completions HTTP/1.1" 400 Bad Request
INFO 06-06 10:12:50 chat_utils.py:332] Detected the chat template content format to be 'string'. You can set `--chat-template-content-format` to override this.
INFO 06-06 10:12:50 logger.py:39] Received request chatcmpl-9743f3531eb04584ad7ff18311ce4d3f: prompt: '<|im_start|>user\n你好<|im_end|>\n<|im_start|>assistant\n', params: SamplingParams(n=1, presence_penalty=0.0, frequency_penalty=0.0, repetition_penalty=1.0, temperature=1.0, top_p=1.0, top_k=-1, min_p=0.0, seed=None, stop=[], stop_token_ids=[], bad_words=[], include_stop_str_in_output=False, ignore_eos=False, max_tokens=1024, min_tokens=0, logprobs=None, prompt_logprobs=None, skip_special_tokens=True, spaces_between_special_tokens=True, truncate_prompt_tokens=None, guided_decoding=None), prompt_token_ids: None, lora_request: None, prompt_adapter_request: None.
INFO:     192.168.72.215:56392 - "POST /v1/chat/completions HTTP/1.1" 200 OK
INFO 06-06 10:12:50 engine.py:275] Added request chatcmpl-9743f3531eb04584ad7ff18311ce4d3f.
INFO 06-06 10:12:50 metrics.py:455] Avg prompt throughput: 0.9 tokens/s, Avg generation throughput: 0.1 tokens/s, Running: 1 reqs, Swapped: 0 reqs, Pending: 0 reqs, GPU KV cache usage: 0.0%, CPU KV cache usage: 0.0%.
INFO 06-06 10:12:55 metrics.py:455] Avg prompt throughput: 0.0 tokens/s, Avg generation throughput: 204.1 tokens/s, Running: 1 reqs, Swapped: 0 reqs, Pending: 0 reqs, GPU KV cache usage: 1.8%, CPU KV cache usage: 0.0%.
INFO 06-06 10:13:05 metrics.py:455] Avg prompt throughput: 0.0 tokens/s, Avg generation throughput: 0.2 tokens/s, Running: 0 reqs, Swapped: 0 reqs, Pending: 0 reqs, GPU KV cache usage: 0.0%, CPU KV cache usage: 0.0%.
INFO 06-06 10:13:15 metrics.py:455] Avg prompt throughput: 0.0 tokens/s, Avg generation throughput: 0.0 tokens/s, Running: 0 reqs, Swapped: 0 reqs, Pending: 0 reqs, GPU KV cache usage: 0.0%, CPU KV cache usage: 0.0%.
INFO 06-06 10:13:48 logger.py:39] Received request chatcmpl-55d7df99f02e466cbf233bf9a2d61802: prompt: '<|im_start|>user\n你好<|im_end|>\n<|im_start|>assistant\n', params: SamplingParams(n=1, presence_penalty=0.0, frequency_penalty=0.0, repetition_penalty=1.0, temperature=0.7, top_p=1.0, top_k=-1, min_p=0.0, seed=None, stop=[], stop_token_ids=[], bad_words=[], include_stop_str_in_output=False, ignore_eos=False, max_tokens=1024, min_tokens=0, logprobs=None, prompt_logprobs=None, skip_special_tokens=True, spaces_between_special_tokens=True, truncate_prompt_tokens=None, guided_decoding=None), prompt_token_ids: None, lora_request: None, prompt_adapter_request: None.
INFO 06-06 10:13:48 engine.py:275] Added request chatcmpl-55d7df99f02e466cbf233bf9a2d61802.
INFO 06-06 10:13:51 metrics.py:455] Avg prompt throughput: 1.8 tokens/s, Avg generation throughput: 91.4 tokens/s, Running: 1 reqs, Swapped: 0 reqs, Pending: 0 reqs, GPU KV cache usage: 0.8%, CPU KV cache usage: 0.0%.
INFO:     192.168.72.215:56428 - "POST /v1/chat/completions HTTP/1.1" 200 OK
INFO 06-06 10:14:03 metrics.py:455] Avg prompt throughput: 0.0 tokens/s, Avg generation throughput: 44.3 tokens/s, Running: 0 reqs, Swapped: 0 reqs, Pending: 0 reqs, GPU KV cache usage: 0.0%, CPU KV cache usage: 0.0%.
INFO 06-06 10:14:13 metrics.py:455] Avg prompt throughput: 0.0 tokens/s, Avg generation throughput: 0.0 tokens/s, Running: 0 reqs, Swapped: 0 reqs, Pending: 0 reqs, GPU KV cache usage: 0.0%, CPU KV cache usage: 0.0%.
INFO:     192.168.72.215:57564 - "POST /v1/embeddings HTTP/1.1" 200 OK
INFO:     192.168.72.215:34178 - "POST /v1/embeddings HTTP/1.1" 200 OK
INFO:     192.168.72.215:34180 - "POST /v1/embeddings HTTP/1.1" 200 OK
INFO:     192.168.72.215:34188 - "POST /v1/embeddings HTTP/1.1" 200 OK
INFO:     192.168.72.215:34190 - "POST /v1/embeddings HTTP/1.1" 200 OK
INFO:     192.168.72.215:34518 - "POST /v1/embeddings HTTP/1.1" 200 OK

**Comment by pengjiang80** [comment]
Qwen3-embedding needs vLLM 0.8.5+. Please set v0.8.5 in advanced - backend version and GPUStack will install vLLM 0.8.5 automatically. Also, you'll need to set --task=embed in backend parameters to make sure it works. Refer to https://github.com/gpustack/gpustack/issues/2170
